lib/logstash/inputs/beats/message_listener.rb

```diff
@@ -119,6 +122,35 @@ def flush_buffer(ctx)
       end
     end
 
+    def extract_tls_peer(hash)
+      if @input.client_authentication_metadata?
+        tls_session = ctx.channel().pipeline().get("ssl-handler").engine().getSession()
+        tls_verified = false
+
+        # throws SSLPeerUnverifiedException if unverified
+        begin
+          tls_session.getPeerCertificates()
+          tls_verified = true
+        rescue SSLPeerUnverifiedException => e
+          if input.logger.debug?
+            input.logger.warn("Failed to get peer certificates.", :exception => e.to_s)
+          end
+        end
+
+        if tls_verified
+          hash['@metadata']['tls_peer'] = {
+            :status     => "verified",
+            :protocol   => tls_session.getProtocol(),
```

Comment:
would this always be `https` ?

Reply:
If you mean `protocol` then no. This is the actual TLS protocol version being used, like `TLSv1.2`

Reply:
doh! wrong plugin, this plugin is not even http based :) 

Is there an option for the cipher chosen too ?

Reply:
Yes, there is! I will add it.